lib/ruby_lsp/server.rb

```diff
@@ -541,7 +551,11 @@ def text_document_diagnostic(message)
         return
       end
 
-      response = @store.cache_fetch(uri, "textDocument/diagnostic") do |document|
+      document = @store.get(uri)
+
+      return send_empty_response(message[:id]) if document.is_a?(ERBDocument)
+
+      response = document.cache_fetch("textDocument/diagnostic") do |document|
```

Comment:
Consider `@store.cache_fetch` is literally just `get(uri).cache_fetch(request_name, &block)`, I don't feel we're losing anything by calling `document.cache_fetch`.
In the contrary, I think the fact that we will need a comment for the condition shows that the proposed implementation is potentially confusing?

I also feel the current implementation fits better with our current convention: if we don't need to go further, we return early.

Reply:
> I also feel the current implementation fits better with our current convention: if we don't need to go further, we return early.

It depends on which convention you are prioritizing. We also don't manipulate the document's cache without going through the store anywhere, which means we'd be breaking a different convention.

In my head, in one case you are changing the implementation of a request, which should be flexible anyway since requests may operate differently and have different requirements. In the other, you are reaching into a lower level API.

It's not a strong opinion and if you feel like the case statement is confusing, feel free to move forward this way.